collector/linux_threads_test.go

```diff
@@ -0,0 +1,29 @@
+package collector
+
+import (
+	"testing"
+)
+
+func TestReadProcessStatus(t *testing.T) {
+	want := 4
+	states, threads, err := getAllocatedThreads()
```

Comment:
To test this properly, we need to change the function to take the path to the proc directory, so that in test we can point to to a fixture directory with some pid directories.

Reply:
Let me figure out with fixture structure and how it works. BTW requested changes have been made